docs: Make README a quickstart and move concept/benchmarking/deploying to separate files (#475)
Unleash/unleash-edge: README.md

[![crates.io](https://img.shields.io/crates/v/unleash-edge?label=latest)](https://crates.io/crates/unleash-edge)
[![Documentation](https://docs.rs/unleash-edge/badge.svg?version=latest)](https://docs.rs/unleash-edge/latest)
![MIT licensed](https://img.shields.io/crates/l/unleash-edge.svg)

> [![Dependency Status](https://deps.rs/crate/unleash-edge/19.2.0/status.svg)](https://deps.rs/crate/unleash-edge/19.2.0)

Reviewer: Is this something we need to manually update every time? If so, can we update `<mostrecentversion>` in our quickstart command below to be the latest version as well?

Ideally we would be able to automate both.

Author: yes, Ideally we'd like to update this before cutting releases, that way the dependency banner matches the version it belongs to, rather than whenever we remembered to update it last.

I'd love to find a way to update this automatically based on most recent version.

For more info on migrating, check out the [migration guide](./migration-guide.md) that details the differences between
Edge and the Proxy and how to achieve similar behavior in Edge.

## Quickstart

> Running Edge in docker with recommended setup

Reviewer: ```suggestion
Running Edge in Docker with our recommended setup:
```

```shell
docker run -it -p 3063:3063 -e STRICT=true -e UPSTREAM_URL=<yourunleashinstance> unleashorg/unleash-edge:<mostrecentversion> edge
```

> ## Deploying Unleash Edge

Reviewer: Maybe Unleash Edge is redundant here? 

```suggestion
## Deploying
```

If you have the [Rust toolchain](https://rustup.rs) installed you can build a binary for the platform you're running by
cloning this repo and running `cargo build --release`. This will give you an `unleash-edge` binary in `./target/release`

> ## Concepts (see [Concepts](./docs/concepts.md))

Reviewer: Given what we did for deployment above, maybe it would be more consistent as this:

```suggestion
## Concepts

See our page on [Edge concepts](./docs/concepts.md)
```

## Performance

> See [Benchmarking](./docs/benchmarking.md)

Reviewer: Consistency nitpicking:

```suggestion
See our page on [Edge benchmarking] [Benchmarking](./docs/benchmarking.md)
```

### Missing metrics in upstream server

#### Possible reasons

- Old edge version. In order to guarantee metrics on newer Unleash versions, you will need to be using Edge v17.0.0 or
  newer

> - Old SDK clients. We've seen some clients, particularly early Python (1.x branch) as well as earlier .NET SDKs (we

Reviewer: ```suggestion
- Old SDK clients. We've noticed that some clients, particularly early Python (1.x branch) as well as earlier .NET SDKs (we
```
